# HG changeset patch
# User welpy-cw <h.w.forms@arcor.de>
# Date 1725008231 -7200
# Node ID c5cada45c8a8ea8bbad7d66cc0b1da40458b70d6
# Parent  6728226a3cfea2c8d83196c568173395f26c663f
Bug 1905912 - Fix persisting of "Grouped By Sort" in virtual folders. r=darktrojan

This basically restores and fixes some supposedly nonsensical and buggy code that had an effect after all.

Differential Revision: https://phabricator.services.mozilla.com/D215499

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -7530,8 +7530,15 @@ var sortController = {
       [newSortType, Ci.nsMsgViewSortOrder.ascending, newSortColumnId],
     ];
     gViewWrapper.showGroupedBySort = false;
     gViewWrapper.endViewUpdate();
+    // Virtual folders don't persist viewFlags well in the back end,
+    // due to a virtual folder being either 'real' or synthetic, so make
+    // sure it's done here.
+    if (gViewWrapper.isVirtual) {
+      gViewWrapper.dbView.viewFlags = gViewWrapper._viewFlags;
+    }
+
     return true;
   },
   reverseSortThreadPane() {
     const grouped = gViewWrapper.showGroupedBySort;
@@ -7599,8 +7606,14 @@ var sortController = {
     // in grouped-by-sort (such as bySize).
     gViewWrapper.beginViewUpdate();
     gViewWrapper.showGroupedBySort = true;
     gViewWrapper.endViewUpdate();
+    // Virtual folders don't persist viewFlags well in the back end,
+    // due to a virtual folder being either 'real' or synthetic, so make
+    // sure it's done here.
+    if (gViewWrapper.isVirtual) {
+      gViewWrapper.dbView.viewFlags = gViewWrapper._viewFlags;
+    }
     threadPane.restoreThreadState(!gViewWrapper.isSingleFolder);
   },
   sortUnthreaded() {
     threadPane.updateListRole(true);
